scripts/routes.js

```diff
@@ -110,6 +147,18 @@ export default function getRoutes(store: Object, plugins: Object[] = []) {
         components={{...common, content: BucketCreatePage}} />
       <Route path="/buckets/:bid/edit"
         components={{...common, content: BucketEditPage}} />
+      <Route path="/buckets/:bid/history"
+        components={{...common, content: BucketHistoryPage}}
+        onEnter={onBucketHistoryEnter.bind(null, store)}
+        onChange={onBucketHistoryEnter.bind(null, store)} />
+      <Route path="/buckets/:bid/collections"
+        components={{...common, content: BucketCollectionsPage}}
+        onEnter={onBucketCollectionsEnter.bind(null, store)}
+        onChange={onBucketCollectionsEnter.bind(null, store)} />
+      <Route path="/buckets/:bid/groups"
+        components={{...common, content: BucketGroupsPage}}
+        onEnter={onBucketGroupsEnter.bind(null, store)}
+        onChange={onBucketGroupsEnter.bind(null, store)} />
```

Comment:
I wonder if we couldn't create some `AuthorizedRoute` higher order component here.